make-subcanon-maps.c

```diff
@@ -0,0 +1,218 @@
+{
+    int i, j, bitPos=0, Gint = 0, bit;
+    
+#if LOWER_TRIANGLE	// Prefer lower triangle to be compatible with Ine Melckenbeeck's Jesse code.
+    for(i=numNodes-1;i>0;i--)
+    {
+        for(j=i-1;j>=0;j--)
+#else   // UPPER_TRIANGLE // this is what we used in the original faye code and paper with Adib Hasan and Po-Chien Chung.
+    for(i=numNodes-2;i>=0;i--)
+    {
+        for(j=numNodes-1;j>i;j--)
+#endif
+        {
+	    if(TinyGraphAreConnected(g,i,j))
+	    {
+		bit = (1 << bitPos);
+		Gint |= bit;
+	    }
+            bitPos++;
+        }
+    }
+    return Gint;
+}
+
+const char *byte_to_binary(int x)
+{
+    static char b[9];
+    b[0] = '\0';
+
+    int z;
+    for (z = 128; z > 0; z >>= 1)
+    {
+        strcat(b, ((x & z) == z) ? "1" : "0");
+    }
+
+    return b;
+}
+
+
+int main(int argc, char* argv[]) {
+    /*
+    read from canon list files
+    for each: 
+        build the G canonical (tinyGraph)
+        remove nodes 1 a a time. and then call tinygraph->whatcanonicalami
+            use induced function called k times. call canon map on induced .
+    */
+    int i, j;
+    k=atoi(argv[1]);
+    _k = k;
+    G = TinyGraphAlloc(k);
+
+    Bk = (1 <<(k*(k-1)/2));
+    char BUF[BUFSIZ];
+    //Create canon list for k
+    sprintf(BUF, CANON_DIR "/canon_list%d.txt", k);
+    FILE *fp_ord=fopen(BUF, "r");
+    assert(fp_ord);
+    int numCanon;
+    fscanf(fp_ord, "%d",&numCanon);
+    int canon_list[numCanon];
+    for(i=0; i<numCanon; i++) fscanf(fp_ord, "%d", &canon_list[i]);
+    fclose(fp_ord);
+
+    // //Create canon list for k-1
+    // sprintf(BUF, CANON_DIR "/canon_list%d.txt", k-1);
+    // FILE *fp_ord=fopen(BUF, "r");
+    // assert(fp_ord);
+    // int numCanon2;
+    // fscanf(fp_ord, "%d",&numCanon);
+    // int canon_list2[numCanon];
+    // for(i=0; i<numCanon; i++) fscanf(fp_ord, "%d", &canon_list[i]);
+    // fclose(fp_ord);
+
+    //Getting and filling canon maps
+    char perm[maxK+1];
+    sprintf(BUF, CANON_DIR "/canon_map%d.bin", k);
+    int Kfd = open(BUF, 0*O_RDONLY);
+    sprintf(BUF, CANON_DIR "/perm_map%d.bin", k);
+    int pfd = open(BUF, 0*O_RDONLY);
+    short int *Kf = Mmap(K, Bk*sizeof(K[0]), Kfd);
+    kperm *Pf = Mmap(Permutations, Bk*sizeof(Permutations[0]), pfd);
+    assert(Kf == K);
+    assert(Pf == Permutations);
+
+    //How to access element from canonmap
+    // TINY_GRAPH *g = TinyGraphAlloc(k-1);
+    // int Gint = TinyGraph2Int(g,k);
+    // int j;
+	// for(j=0;j<k;j++) perm[j]=0;
+	// ExtractPerm(perm, Gint);
+	// //printf("K[%d]=%d [%d];", Gint, K[Gint], canon_list[K[Gint]]);
+	// printf("%d", K[Gint]); // Note this is the ordinal of the canonical, not its bit representation
+	// for(j=0;j<k;j++) printf(" %d", Varray[(unsigned)perm[j]]);
+	// puts("");
+
+    /*
+        For every canonical in canonical_list
+            create tinygraph
+            print canonical
+            
+            For every induced subgraph of size k-1
+                get canonical from k-1

```

Comment:
Is this the size k canonical that represents the k-1 subgraph, or the (k-1)-sized canonical?  Now that you're forcing me to think about this, I'm not sure which we should use... will need to think about this some more.